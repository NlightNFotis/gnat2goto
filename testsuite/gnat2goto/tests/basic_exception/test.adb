procedure Test is
   Too_Small : exception;
   procedure More_Than_Five (Num : Integer) is
   begin
      if Num <= 5 then
         raise Too_Small with "way too small!";
      end if;
   end More_Than_Five;

   Error_Count : Integer := 0;
begin
   begin
      More_Than_Five (10);
   exception
      when Too_Small =>
         Error_Count := Error_Count + 1;
   end;
   begin
      More_Than_Five (5);
   exception
      when Too_Small =>
         Error_Count := Error_Count + 1;
   end;
   --  this will be unreachable due to implementation of `raise`
   pragma Assert (Error_Count = 1);
end Test;
