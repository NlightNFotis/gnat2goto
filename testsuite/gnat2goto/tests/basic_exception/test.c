#include <assert.h>

int foo(unsigned char* arg)
{
   if (arg[0] == 't')
      return 1;
   else
      return 0;
}

int main()
{
   int res = foo((unsigned char *)"test");
   assert(res == 0);
   return 0;
}
