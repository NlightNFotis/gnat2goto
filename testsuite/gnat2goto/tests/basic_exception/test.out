[assertion.2] file test.adb line 6 Ada Exception: FAILURE
[assertion.1] file test.adb line 16 Ada Check assertion: SUCCESS
[test.assertion.1] line 25 assertion Error_Count = 1: SUCCESS
VERIFICATION FAILED
